core/src/main/java/org/infinispan/functional/impl/EntryViews.java

```diff
@@ -280,24 +314,33 @@ public String toString() {
    }
 
    private static final class EntryAndPreviousReadWriteView<K, V> implements ReadWriteEntryView<K, V> {
-      final CacheEntry<K, V> entry;
+      final CacheEntry entry;
       final V prevValue;
       final Metadata prevMetadata;
+      private final CacheEncoders cacheEncoders;
 
-      private EntryAndPreviousReadWriteView(CacheEntry<K, V> entry, V prevValue, Metadata prevMetadata) {
+      private EntryAndPreviousReadWriteView(CacheEntry<K, V> entry,
+                                            V prevValue,
+                                            Metadata prevMetadata,
+                                            CacheEncoders cacheEncoders) {
          this.entry = entry;
          this.prevValue = prevValue;
          this.prevMetadata = prevMetadata;
+         this.cacheEncoders = cacheEncoders;
       }
 
       @Override
       public K key() {
-         return entry.getKey();
+         return (K) cacheEncoders.keyFromStorage(entry.getKey());
       }
 
       @Override
       public Optional<V> find() {
-         return Optional.ofNullable(prevValue);
+         return Optional.ofNullable((V) cacheEncoders.valueFromStorage(prevValue));
+      }
+
+      public V getCurrentValue() {
+         return (V) cacheEncoders.valueFromStorage(entry.getValue());
```

Comment:
This is used for snapshot. the getValue gives PrevValue there, and I need the newValue encoded.